lib/Common/Memory/RecyclerPointers.h

```diff
@@ -468,10 +489,41 @@ const T& max(const T& a, const NoWriteBarrierField<T>& b) { return a > b ? a : b
 template<class T> inline
 const T& max(const NoWriteBarrierField<T>& a, const NoWriteBarrierField<T>& b) { return a > b ? a : b; }
 
-template<typename T, typename Comparer>
-void qsort_s(Memory::WriteBarrierPtr<T>* _Base, size_t _NumOfElements, size_t _SizeOfElements, Comparer comparer, void* _Context)
+// QuickSort Array content
+//
+template <class Policy>
+struct _QuickSortImpl
+{
+    template<class T, class Comparer>
+    static void qsort_s(T* arr, size_t count, const Comparer& comparer, void* context)
+    {
+        // by default use system qsort_s
+        ::qsort_s(arr, count, sizeof(T), comparer, context);
```

Comment:
> but at the moment not fine tuned, so would like to use system version if available.

@jianchun The biggest issue with system qsort_r is that the problems are hard to track once we hit. I feel your reasoning and makes sense.

How about if we create a Github issue OR add todo there -> `Make our quicksort great and use it instead of system qsort_r because it is unreliable cross platform. It really is..`  ?